sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1097,4 +1097,9 @@ private[spark] object QueryCompilationErrors {
       s"and column $colName cannot be resolved. Expected $expectedNum columns named $colName but " +
       s"got ${actualCols.map(_.name).mkString("[", ",", "]")}")
   }
+
+  def numberOfPartitionsNotAllowedWithUnspecifiedDistributionError(): Throwable = {
+    throw new AnalysisException("The number of partitions can't be specified with unspecified" +
```

Comment:
this shouldn't be throw here

Reply:
Nice finding. Would you mind to submit a MINOR PR fixing this? I can also work on this as well.